packages/context/src/binding.ts

```diff
@@ -246,11 +262,15 @@ export class Binding<T = BoundValue> {
         }
       }
     }
+    optionsOrSession = asResolutionOptions(optionsOrSession);
```

Comment:
```suggestion
    const options = asResolutionOptions(optionsOrSession);
```

After this line, we always have the options object, right?